### PR TITLE
Linting errors resolved.(excluding max. line length error)

### LINKED ISSUE #99: Need to fix lint errors
## Description
As a coder,
I need to fix the lint errors to ensure good code quality and readability.
so that it becomes easier to collaborate and scale.

## Mocks
Running "ng lint" listed 100s of such lint errors.
many

So I ran "ng lint --fix" to fix the minor errors:
rsz_fix

But errors like, " 'ButtonNavComponent' should have prefix 'app' ", Exceeding maximum line width etc. would need to be fixed manually.
rsz_not-fixed

### Update [Required]
- [ ] All these lint errors fixed.
### Enhancement to Update [Optional]
- [ ] In future, whenever somebody commits, he/she should do a lint check first before committing. This would help in maintaining some code consistency.  So the Readme.md should be updated.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1-2 hours

### PR DESCRIPTION
**Pull request by lunayach**
**Description:**
I tried to fix lint errors EXCEPT, for the one which accounts for (maximum line length > 140)
Fixes #99 
**Type of Change:**
- Quality Assurance

**Code/Quality Assurance Only**

- Bug fix (a non-breaking change which fixes an issue)
### How Has This Been Tested?
I ran `ng lint`  to test it, and it said 

> all tests passed.

### Checklist:

- [x] I have performed a self-review of my own code or materials

- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

File: config/settings.js [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-    },
-    databaseSetup: {
-        DATABASE_NAME: 'pcprepkit',
-        USERNAME: 'pcprepkit',
-        PASSWORD: 'pcprepkit'
-    }
+  basePath: 'http://localhost:3000/',
+  googleAuth: {
+    clientID: '311448429695-va7g4f7takj7blpudu4gtvf9qeqr86a0.apps.googleusercontent.com',
+    clientSecret: 'YG0pAqQwrR8Lsdyh89cFHg3x',
```
Comment (Raxerz) [inline config/settings.js]: Was there a need to change the `clientID` and `clientSecret`? If not, then please revert the changes made to these keys.

File: config/settings.js [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
+  },
+  secretKey: 'pcprepkittokengenkey',
+  nodeMailer: {
+    PROVIDER: 'Gmail',
+    EMAIL: 'pcprepkit@gmail.com',
+    PASSWORD: 'pcprepkit9'
+  },
+  apiai: {
+    clientToken: '0f67eb54f1d74ea4a3d9c34a76da68ca'
+  },
```
Comment (Raxerz) [inline config/settings.js]: Again, was there a need to change the `clientToken`? Please revert this change if it was not necessary

File: config/settings.js [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
+  databaseSetup: {
+    DATABASE_NAME: 'pcprepkit',
+    USERNAME: 'root',
+    PASSWORD: '96678239'
+  }
```
Comment (Raxerz) [inline config/settings.js]: Please revert the changes made to the database config as this may affect the others using the app.
Also, ensure that such changes that can impact others are not made or excluded from the PR.

File: src/app/app.component.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { SharedDataService } from './services/shared.data.service';
 export class AppComponent {
     private static _localStorageKey = 'pcprepkitUser';
     title = 'PC Prep Kit';
-    public position = 'col-md-10 col-md-offset-2';
+    public position = 'col-mat-10 col-mat-offset-2';
     public loggedIn = false;
```
Comment (Raxerz) [inline src/app/app.component.ts]: Why are there material related changes in a PR for linting fixes?

File: config/passport.js [hunk 5]
```diff
@@ -8,6 +8,7 @@ module.exports = function(passport, models) {
     const localUser = models.user_account;
     const progress = models.progress;
     let fname, lname;
+  passport.authenticate('oath2', {failWithError: true});
```
Comment (Raxerz) [inline config/passport.js]: What is this change?

**Comment by lunayach** [comment]
@Raxerz I think we should add the settings.js in gitignore . So that changes we do locally are not propagated in commits.

**Comment by Raxerz** [comment]
@lunayach Please fix all the issues reported by the codacy-bot

**Comment by Raxerz** [comment]
@lunayach It actually is a good idea to add the settings.js file to .gitignore but there may be cases when a new feature which requires an id, secret or api key of some 3rd party provider to be added to the settings.js file, then it will have to be removed from the .gitignore file to commit and added back to it later, this might cause issues if it's not removed or added back appropriately. Also, some might not even be aware of the file being added to .gitignore.
These are just a couple of my thoughts on this, maybe Shreyans and Paavini can give their opinions on it too. We can get to a collective decision after that.

**Comment by Raxerz** [comment]
@lunayach I forgot to mention that it is required that you squash your commits to a single commit for the PRs you make. The idea is to map a single commit with the issue.

**Comment by lunayach** [comment]
@Raxerz I think in the Readme.md we can mention about the gitignore. 
For the "Squash commit" thing, I'll keep in mind the next time. :+1: 
And is this codacy-bot thing configurable, because one of the thing either our lint or this has to change the rules for default indention (viz. spaces, tabs) etc.

**Comment by shreyans29** [comment]
@lunayach  Apart from what Rajat mentioned, please follow the PR format.  you will find it in the master branch, update the code in your branch with latest develop code and that should make travis pass.

File: package.json [hunk 6]
```diff
@@ -32,6 +32,7 @@
     "core-js": "^2.4.1",
     "crypto": "0.0.3",
     "debug": "~2.6.3",
+    "email-verify": "^0.2.1",
```
Comment (shreyans29) [inline package.json]: Is the email-verify module functioning as expected ?

**Comment by Raxerz** [comment]
@lunayach I am not sure if the codacy-bot is configurable, I have read only access to codacy so that limits any sort of experiments I can do with it. 
Also the issues reported by codacy needs to be fixed for the codacy review to pass.
Please make the necessary changes.

File: package.json (hunk 6 above)
Comment (lunayach) [reply package.json]: I think it's not. It was giving status 400 on my college mail ID.

**Comment by lunayach** [comment]
Okay, I would try to see if I can tweak the lint config file for default indent or would have to do these manually.

**Comment by Raxerz** [comment]
@lunayach Can you fix the remaining lint error so that we can merge this PR?
